src/btree/bt_split.c

```diff
@@ -586,6 +586,16 @@ __split_deepen(WT_SESSION_IMPL *session, WT_PAGE *parent, uint32_t children)
 	 * fails, we don't roll back that change, because threads may already
 	 * be using the new index.
 	 */
+	for (parent_refp = pindex->index, i = 0;
```

Comment:
Hi @agorrod, that macro walks the index that is currently attached to an internal page.  At this point in the split code, we have already swapped in a new index, and we are walking the old one here to schedule any instantiated keys to be freed.